tests/typecheck/core/test_checks.yml

```diff
@@ -0,0 +1,24 @@
+- case: test_checks_register
+  main: |
+    from typing import Any, Sequence, Optional
+
+    from django.core.checks import register, Warning, CheckMessage
+    from django.apps.registry import Apps
+
+
+    @register("foo", deploy=True)
+    def check_foo(app_configs: Apps, databases: Optional[Sequence[str]], **kwargs: Any) -> list[Warning]:
+        if databases and 'databass' in databases:
+            return [Warning("Naughty list")]
+        return []
```

Comment:
most checks don't take a `databases` argument, so I'd like a test that checks that situation too